src/components/molecules/ChatMessageBox/ChatMessageBox.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useState, useEffect } from "react";
+import { useForm } from "react-hook-form";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faPaperPlane } from "@fortawesome/free-solid-svg-icons";
+
+import "./ChatMessageBox.scss";
+
+export interface ChatMessageBoxProps {
+  sendMessage: (text: string) => void;
+}
+
+export const ChatMessageBox: React.FC<ChatMessageBoxProps> = ({
+  sendMessage,
+}) => {
+  const [isSendingMessage, setMessageSending] = useState(false);
+
+  // This logic dissallows users to spam into the chat. There should be a delay, between each message
+  useEffect(() => {
+    if (isSendingMessage) {
+      setTimeout(() => {
+        setMessageSending(false);
+      }, 500);
+    }
+  }, [isSendingMessage]);
+
+  const { register, handleSubmit, reset, watch } = useForm<{
+    message: string;
+  }>({
+    mode: "onSubmit",
+  });
+
+  const onSubmit = handleSubmit(({ message }) => {
+    setMessageSending(true);
+    sendMessage(message);
+    reset();
+  });
```

Comment:
Also, do we want to ensure that `message` actually contains a message here? I know we disable the send button below already. But this would allow us to be extra sure/explicit about the expected state here.

Reply:
I'm not working on ChatMessage at all. This code is just moved into a separate file with the same (working) logic. 

Do we actually need chat message box refactoring? Should it be a part of this PR?

Reply:
Personally when it's like a 1 or 2 line change in code that is already being touched, I opt to fix it since it takes more time/effort to discuss/write up an issue to resolve it in future/etc than it would to just solve it then and there.

@mike-lvov often has different opinions. He can make the final call here.

Reply:
@mike-lvov Do we need to refactor Chatmessage in scope of this PR? Looks like it's huge refactoring, I'm not sure it's the best place to refactor Chat

Reply:
> Looks like it's huge refactoring

@yurii-lubynets Huge how? It would be changing a couple of variable names/etc scoped within a single component?

Reply:
@yurii-lubynets @0xdevalias given that it's only a few changes(var name change + small useEffect refactor), I'm OK with having it in this PR

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/1343/commits/f1aa24f7ec8d8d83f24d76aefa459cb1e12080d6